Fix version resolution in  CLI

gsmet opened the pull request:
This was broken in #49926 where the property name was changed mistakenly.

To be honest, I'm also not sure the new method is as safe as it used to be, especially on Windows as we also dropped the specific handling of the jar pattern. There was a specific comment about it in the preexisting method.
I suppose we will see how it goes...

gsmet committed:
Fix version resolution in  CLI

This was broken in #49926 where the property name was changed
mistakenly.

To be honest, I'm also not sure the new method is as safe as it used to
be, especially on Windows as we also dropped the specific handling of
the jar pattern. There was a specific comment about it in the
preexisting method.
I suppose we will see how it goes...

aloubyansky commented on devtools/cli-common/src/main/java/io/quarkus/cli/common/VersionHelper.java:
```diff
@@ -34,7 +34,7 @@ public static String clientVersion() {
                 }
             }
-            version = props.getProperty("quarkus.version", "999-SNAPSHOT");
+            version = props.getProperty("quarkus-core-version", "999-SNAPSHOT");
```
Perhaps it should be something like `N/A` or `unknown`, instead of `999-SNAPSHOT`.

gsmet replied:
But I would prefer that @iocanel sheds some light on why he changed the property and why he's not throwing an error anymore.

gsmet committed:
Bring back the original implementation of getting version in CLI

I think the main issue that Ioannis tried to work around was an artifact
missing.
This should handle the problem properly.
